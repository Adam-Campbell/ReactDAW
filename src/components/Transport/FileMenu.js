import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as ActionCreators from '../../actions';
import { modalTypes } from '../../constants';

export const FileMenu = props => {

    const handleSave = () => {
        if (props.saveName) {
            const namespacedSaveName = `[track_name] ${props.saveName}`;
            props.saveState(namespacedSaveName);
        } else {
            props.openModal(modalTypes.saveAs);
        }
    }

    return (
        <div 
            className="menu__container" 
            ref={props.containerRef}
            onClick={props.toggleMenu}
        >
            <p 
                className="menu__label"
            >File</p>
            <ul 
                className={`menu__list ${props.isOpen ? 'menu__list--visible' : ''}`}
            >
                <li 
                    className="menu__item"
                    onClick={() => props.openModal(modalTypes.open)}
                >
                    <p className="menu__item-text">Open</p>
                </li>
                <li 
                    className="menu__item"
                    onClick={handleSave}
                >
                    <p className="menu__item-text">Save</p>
                </li>
                <li 
                    className="menu__item"
                    onClick={() => props.openModal(modalTypes.saveAs)}
                >
                    <p className="menu__item-text">Save As</p>
                </li>
            </ul>
        </div>
    );
}

FileMenu.propTypes = {
    isOpen: PropTypes.bool.isRequired,
    toggleMenu: PropTypes.func.isRequired,
    containerRef: PropTypes.object.isRequired
}

const mapStateToProps =  state => ({
    saveName: state.main.present.composition.saveName
});

export default connect(
    mapStateToProps,
    { 
        openModal: ActionCreators.openModal,
        saveState: ActionCreators.saveState
    }
)(FileMenu);