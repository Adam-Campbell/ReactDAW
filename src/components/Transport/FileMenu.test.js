import React from 'react';
import { FileMenu } from './FileMenu';
import { shallow } from 'enzyme';

test('renders closed state correctly', () => {
    const component = shallow(
        <FileMenu 
            isOpen={false}
            toggleMenu={jest.fn()}
            containerRef={React.createRef()}
        />
    );
    expect(component).toMatchSnapshot();
});

test('renders open state correctly', () => {
    const component = shallow(
        <FileMenu 
            isOpen={true}
            toggleMenu={jest.fn()}
            containerRef={React.createRef()}
        />
    );
    expect(component).toMatchSnapshot(); 
});