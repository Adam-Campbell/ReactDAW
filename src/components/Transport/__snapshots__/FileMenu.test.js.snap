// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders closed state correctly 1`] = `
<div
  className="menu__container"
  onClick={[MockFunction]}
>
  <p
    className="menu__label"
  >
    File
  </p>
  <ul
    className="menu__list "
  >
    <li
      className="menu__item"
    >
      <p
        className="menu__item-text"
      >
        New
      </p>
    </li>
    <li
      className="menu__item"
      onClick={[Function]}
    >
      <p
        className="menu__item-text"
      >
        Open
      </p>
    </li>
    <li
      className="menu__item"
      onClick={[Function]}
    >
      <p
        className="menu__item-text"
      >
        Save
      </p>
    </li>
    <li
      className="menu__item"
      onClick={[Function]}
    >
      <p
        className="menu__item-text"
      >
        Save As
      </p>
    </li>
  </ul>
</div>
`;

exports[`renders open state correctly 1`] = `
<div
  className="menu__container"
  onClick={[MockFunction]}
>
  <p
    className="menu__label"
  >
    File
  </p>
  <ul
    className="menu__list menu__list--visible"
  >
    <li
      className="menu__item"
    >
      <p
        className="menu__item-text"
      >
        New
      </p>
    </li>
    <li
      className="menu__item"
      onClick={[Function]}
    >
      <p
        className="menu__item-text"
      >
        Open
      </p>
    </li>
    <li
      className="menu__item"
      onClick={[Function]}
    >
      <p
        className="menu__item-text"
      >
        Save
      </p>
    </li>
    <li
      className="menu__item"
      onClick={[Function]}
    >
      <p
        className="menu__item-text"
      >
        Save As
      </p>
    </li>
  </ul>
</div>
`;
